helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -141,6 +157,53 @@ public boolean create(String path, T record, int options) {
     return result._retCode == RetCode.OK;
   }
 
+  /**
+   * Creates a ZkBaseDataAccessor with a custom implementation of ZkSerializer.
+   * If dedicated flag is set to true, it will use a dedicated ZkConnection, which allows ephemeral
+   * node creation, callbacks, and session management.
+   * If dedicated flag is set to false, it will use a shared ZkConnection, which only allows simple
+   * CRUD operations to ZooKeeper.
+   * @param zkAddress
+   * @param zkSerializer
+   * @param dedicated
+   */
+  public ZkBaseDataAccessor(String zkAddress, ZkSerializer zkSerializer, boolean dedicated) {
```

Comment:
Just my 2 cents: to me, passing a boolean param to a method to determine the behavior is kind of code smell.
See
* https://softwareengineering.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior
* https://martinfowler.com/bliki/FlagArgument.html

Can we have a better design for this? In my opinion, at least, we can create an enum and pass it:
```
enum ZkClientType {
  DedicatedZkClient, SharedZkClient;
}

ZkBaseDataAccessor(String zkAddress, ZkSerializer zkSerializer, ZkClientType type);
```
I think this would look better than this ugly boolean flag..

Reply:
Sure. I agree with this suggestion. Let me define an enum.

Reply:
Enum added.